src/components/ChannelsAvailabilityDropdown/ChannelsAvailabilityDropdown.tsx

```diff
@@ -1,140 +1,69 @@
+import { useIntl } from "react-intl";
 
-import { useStyles } from "./styles";
-
-type Channels = Pick<
-  CollectionList_collections_edges_node_channelListings,
-  "isPublished" | "publicationDate" | "channel"
->;
+import ChannelsAvailabilityMenuContent from "../ChannelsAvailabilityMenuContent";
+import { messages } from "./messages";
+import {
+  CollectionChannels,
+  getChannelAvailabilityColor,
+  getChannelAvailabilityLabel,
+  getDropdownColor
+} from "./utils";
 
 export interface ChannelsAvailabilityDropdownProps {
-  allChannelsCount: number;
-  channels: Channels[];
-  showStatus?: boolean;
+  channels: CollectionChannels[] | null;
 }
 
-const isActive = (channelData: Channels) => channelData?.isPublished;
-
 export const ChannelsAvailabilityDropdown: React.FC<ChannelsAvailabilityDropdownProps> = ({
-  allChannelsCount,
-  channels,
-  showStatus = false
+  channels
 }) => {
   const intl = useIntl();
-  const classes = useStyles({});
-  const localizeDate = useDateLocalize();
-  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [isPopupOpen, setPopupOpen] = React.useState(false);
+  const anchor = React.useRef<HTMLDivElement>(null);
 
-  const handleClick = event => setAnchorEl(event.currentTarget);
-  const handleClose = () => setAnchorEl(null);
-  const activeInAllChannels = React.useMemo(
-    () => showStatus && channels.every(isActive),
-    [channels, showStatus]
-  );
+  const dropdownColor = React.useMemo(() => getDropdownColor(channels), [
+    channels
+  ]);
+
+  const handleMouseOver = () => setPopupOpen(true);
+  const handleMouseLeave = () => setPopupOpen(false);
+
+  if (!channels || channels.length === 0) {
+    return (
+      <Pill label={intl.formatMessage(messages.noChannels)} color="error" />
+    );
+  }
 
   return (
-    <div onClick={e => e.stopPropagation()}>
-      <div
-        aria-controls="availability-menu"
-        aria-haspopup="true"
-        role="button"
-        onClick={handleClick}
-      >
-        <StatusLabel
-          label={intl.formatMessage(
-            {
-              defaultMessage: "{count}/{allCount} channels",
-              description: "product status title"
-            },
-            {
-              allCount: allChannelsCount,
-              count: channels.length
-            }
-          )}
-          status={
-            showStatus ? (activeInAllChannels ? "success" : "error") : undefined
-          }
+    <div
+      onClick={e => {
+        e.preventDefault();
+        e.stopPropagation();
+      }}
+      ref={anchor}
+      onMouseOver={handleMouseOver}
+      onMouseLeave={handleMouseLeave}
+    >
+      <div aria-controls="availability-menu" aria-haspopup="true" role="button">
+        <Pill
+          label={intl.formatMessage(messages.dropdownLabel, {
+            channelCount: channels.length
+          })}
+          color={dropdownColor}
+          onClick={() => null} // required for dashed border
```

Comment:
Why is this component requiring us to pass onClick when the usage for such styling is wider than just clickable pill?

Reply:
@dominik-zeglen We agreed with @pwgryglak that pills with poppers work better onHover than onClick, yet they still have to be interactive (with dashed border and hover state). Can we refactor pills so that it is possible to add dashed border without passing onClick?
Right now `clickable` css class is only applied when `onClick` is passed